android/app/src/main/kotlin/com/meridianaprs/meridian_aprs/ClassicBtChannel.kt

```
@@ -0,0 +1,241 @@
+                    .createRfcommSocketToServiceRecord(SPP_UUID)
+            } catch (e: Exception) {
+                emitState("error", e.message ?: "socket create failed")
+                return@Thread
+            }
+            try {
+                sock.connect()
+            } catch (e: IOException) {
+                try { sock.close() } catch (_: IOException) {}
+                emitState("error", e.message ?: "connect failed")
+                return@Thread
+            } catch (e: SecurityException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)